### PR TITLE
Sync develop changes 8/27 - 9/11 to hdf5_1_14.

### PR DESCRIPTION
**Pull request by lrknox**

### CODE REVIEW

File: config/cmake/hdf5-config.cmake.in [hunk 1]
```diff
@@ -91,7 +91,7 @@ if (${HDF5_PACKAGE_NAME}_ENABLE_PARALLEL)
   find_package(MPI QUIET REQUIRED)
 endif ()
 
-if (${HDF5_PACKAGE_NAME}_ENABLE_THREADSAFE OR ${HDF5_PACKAGE_NAME}_ENABLE_SUBFILING_VFD)
+if (${HDF5_PACKAGE_NAME}_ENABLE_THREADS)
```
Comment (lrknox) [inline config/cmake/hdf5-config.cmake.in]:
```suggestion
if (${HDF5_PACKAGE_NAME}_ENABLE_THREADSAFE OR ${HDF5_PACKAGE_NAME}_ENABLE_SUBFILING_VFD)
```

**Review by hyoklee — APPROVED** [review]

File: doc/parallel-compression.md [hunk 2]
```diff
@@ -332,3 +332,20 @@ H5Pset_libver_bounds(fapl_id, H5F_LIBVER_LATEST, H5F_LIBVER_LATEST);
 hid_t file_id = H5Fcreate("file.h5", H5F_ACC_TRUNC, H5P_DEFAULT, fapl_id);
 ...
 ```
+
+[u1]: https://github.com/HDFGroup/hdf5/blob/develop/HDF5Examples/C/H5PAR/ph5_filtered_writes.c
```
Comment (jhendersonHDF) [inline doc/parallel-compression.md]: Should all of these links point to a 1.14 version in this branch?

File: release_docs/RELEASE_PROCESS.md [hunk 3]
```diff
@@ -161,3 +161,19 @@ For more information on the HDF5 versioning and backward and forward compatibili
 ### 11. Conduct Release Retrospective (Release Manager)
 1. Schedule time and solicit comments from retrospective
 2. Identify issues and document them
+
+[u1]: https://github.com/HDFGroup/hdf5/blob/develop/release_docs/RELEASE.txt
```
Comment (jhendersonHDF) [inline release_docs/RELEASE_PROCESS.md]: Same comment here about using 1.14 versions of the links rather than develop

**Review by byrnHDF — APPROVED** [review]

File: release_docs/RELEASE_PROCESS.md (hunk 3 above)
Comment (byrnHDF) [reply release_docs/RELEASE_PROCESS.md]: Yes

File: doc/parallel-compression.md (hunk 2 above)
Comment (byrnHDF) [reply doc/parallel-compression.md]: Yes

File: doc/parallel-compression.md [hunk 4]
```diff
@@ -332,3 +332,20 @@ H5Pset_libver_bounds(fapl_id, H5F_LIBVER_LATEST, H5F_LIBVER_LATEST);
 hid_t file_id = H5Fcreate("file.h5", H5F_ACC_TRUNC, H5P_DEFAULT, fapl_id);
 ...
 ```
+
+[u1]: https://github.com/HDFGroup/hdf5/blob/hdf5_1_14/HDF5Examples/C/H5PAR/ph5_filtered_writes.c
+[u2]: https://github.com/HDFGroup/hdf5/blob/hdf5_1_14/HDF5Examples/C/H5PAR/ph5_filtered_writes_no_sel.c
+[u3]: https://hdfgroup.github.io/hdf5/develop/group___h5_d.html#gaf6213bf3a876c1741810037ff2bb85d8
```
Comment (jhendersonHDF) [inline doc/parallel-compression.md:338]: Shouldn't all of these documentation links below also be updated to point to the 1.14 docs rather than develop?

Comment (byrnHDF) [reply doc/parallel-compression.md:338]: That file has a number of problems that I'm addressing in a followup PR

**Pull request merged**